sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -84,4 +117,28 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
             return Flux.fromIterable(elements);
         }).subscribe(coreSubscriber);
     }
+
+    private Flux<FeedResponse<T>> byPage(CosmosPagedFluxOptions pagedFluxOptions, Context context) {
+        final AtomicReference<Context> parentContext =  new AtomicReference<>(Context.NONE);
+
+        return this.optionsFluxFunction.apply(pagedFluxOptions).doOnSubscribe(ignoredValue -> {
+            if (pagedFluxOptions.getTracerProvider().isEnabled()) {
+                reactor.util.context.Context reactorContext = FluxUtil.toReactorContext(context);
+                Objects.requireNonNull(reactorContext.hasKey(TracerProvider.MASTER_CALL));
+                Optional<Object> callerFunc = reactorContext.getOrEmpty(TracerProvider.NESTED_CALL);
+                if (!callerFunc.isPresent()) {
+                    parentContext.set(pagedFluxOptions.getTracerProvider().startSpan(pagedFluxOptions.getTracerSpanName(),
+                        context.addData(TracerProvider.ATTRIBUTE_MAP, pagedFluxOptions.getTracingAttributes()), ProcessKind.DATABASE));
+                }
+            }
+        }).doOnEach(responseSignal -> {
```

Comment:
Ending span on very first element of pagedFlux , avoiding accountability of any user side delay. Please let me know if anyone think over wise and want it on OnComplete()

Reply:
@trask Would like to know your thoughts on the expected behavior, if we should end the span of the first element received.

Reply:
@simplynaveen20 @samvaity That's a great question! I think that ideally it would capture until complete, while at the same time somehow capturing the internal time that was spent waiting on cosmosdb. Maybe the diagnostic data returned from the cosmosdb server could represent the internal time, as event(s) on the overarching span? This doesn't have to be part of the initial work though, and it's ok if you want to end the span on first element until we have a way to separate cosmosdb internal time from (external) user time while iterating over results.

Reply:
Moved to to doOnComplete()(It will include both sdk + user side time). However we have diagnostics to find out exact time taken by SDK

Reply:
oh, nice, will that be part of the diagnostics that we capture in the next phase?